File: codepush.rb
require 'rest-client'
require 'json'
require "highline/import"

$circle_token = "248e173620570e0753cc40bd14d0ccde06a5c0ac"

def codepush(platform, branch, mandatory)

  branch = "master" if branch.nil? || branch.empty?

  response = RestClient.post(
  "https://#{$circle_token}:@circleci.com/api/v1.1/project/github/phorest/rn-branded-app/tree/#{branch}",
  {build_parameters: { CIRCLE_JOB: "code_push", PLATFORM: platform, MANDATORY: mandatory}}.to_json,
  {'accept': 'application/json',  "Content-Type": 'application/json'})
  circle_job = JSON.parse(response.body)
  build_url = circle_job['build_url']

  puts "New #{platform} build at #{build_url}."
  system("open #{build_url}")
end

puts "New Branded App CodePush QA."
puts "\n"

branch = ask "Please enter the branch you want to code push (hit enter for master): "
mandatory = ask "Is this a mandatory release (true | false): "

codepush("ios", branch, mandatory)
codepush("android", branch, mandatory)

File: commands/codepush.js
const axios = require("axios");

module.exports = function codepush(data) {
  const request = data.text.split(" ");
  const branch = request[2];
  const mandatory = request[3];
  const platform = request[4];
  return codepushToQA(branch, mandatory, platform).then((response) => {
    return response;
  });
};

async function codepushToQA(branch, mandatory, platform) {
  console.log(branch, mandatory, platform);
  return axios
    .post(
      `https://${process.env.CIRCLE_TOKEN}:@circleci.com/api/v1.1/project/github/phorest/rn-branded-app/tree/${branch}`,
      {
        build_parameters: {
          CIRCLE_JOB: "code_push",
          PLATFORM: platform,
          MANDATORY: mandatory
        }
      }
    )
    .then(function (response) {
      console.log(JSON.stringify(response.data.build_url));
      return `Ciao, Codepushing to QA from branch ${branch} with mandatory: ${mandatory} for platform: ${platform} now. \n ${response.data.build_url}`;
    })
    .catch(function (error) {
      console.log(error);
      return "codepush failed";
    });
}
